### PR TITLE
Preview mode

### PR DESCRIPTION
**Pull request by roperzh**
**What**

This adds logic to handle the preview mode, as described [here](https://haiku.quip.com/aWHpAEvMmR9N)

[Trello card](https://trello.com/c/79CV50wA/392-build-rudimentary-preview-mode)

**Comments**

- Besides the logic described in the Quip document we are also handling a bunch of edge cases and constraints like:
  - Copy/Cut/Paste are disabled on preview mode
  - Asset instantiation is disabled
  - Several keyboard shortcuts/interactions
- Some events interactions don't work properly (mouseover, keypress, etc) but they don't work properly in master too, they are being tackled by @roperzh in a separate PR as part of the events UI.
- I created a [namespace in `haiku-common`](https://github.com/HaikuTeam/mono/compare/preview-mode?expand=1#diff-fa3e0b219a520917f822eb2fd81942ec) to define interaction modes, the only place where they are still hardcoded is in the player, but I need to review with @matthewtoast if this is OK.

preview-mode-2.m4v GIF

### CODE REVIEW

**Comment by roperzh** [comment]
Including @taylorpoe to review the UI/Design:

screen shot 2017-11-07 at 20 43 15

screen shot 2017-11-07 at 20 43 08

toggle

**Review by taylorpoe — CHANGES_REQUESTED** [review]
@roberzh as much as I love the toggle, I think the result is too text heavy. I think we should try something more minimal. I think this should work well:
eyeball

**Comment by taylorpoe** [comment]
@roperzh eyeball svg icon is in the linearicons collection, which I believe you have.

File: packages/haiku-common/config/experiments.json [hunk 1]
```diff
@@ -8,5 +8,9 @@
     "development": true,
     "staging": true,
     "production": true
+  },
+  "PreviewMode": {
+    "test": true,
+    "development": true
```
Comment (stristr) [inline packages/haiku-common/config/experiments.json:14]: Note: after #83 is merged this will simplify to `"development": true`.

**Pull request merged**